sound/soc/intel/boards/jsl_da7219_max98373.c

```diff
@@ -0,0 +1,421 @@
+// SPDX-License-Identifier: GPL-2.0
+// Copyright(c) 2019 Intel Corporation.
+
+/*
+ * Intel Jasperlake I2S Machine driver for DA7219 + MAX98373 codec
+ */
+
+#include <linux/dmi.h>
+#include <linux/input.h>
+#include <linux/module.h>
+#include <linux/platform_device.h>
+#include <sound/jack.h>
+#include <sound/pcm.h>
+#include <sound/pcm_params.h>
+#include <sound/soc.h>
+#include <sound/soc-acpi.h>
+#include "../../codecs/da7219.h"
+#include "../../codecs/da7219-aad.h"
+#include "../../codecs/hdac_hdmi.h"
+#include "../common/soc-intel-quirks.h"
+
+#define DIALOG_CODEC_DAI	"da7219-hifi"
+#define MAX98373_CODEC_DAI	"max98373-aif1"
+#define MAXIM_DEV0_NAME		"i2c-MX98373:00"
+#define MAXIM_DEV1_NAME		"i2c-MX98373:01"
+
+#define SOF_DA7219_MCLK_24MHZ	BIT(0)
+
+/* Default: MCLK 24M */
+static unsigned long sof_da7219_quirk = SOF_DA7219_MCLK_24MHZ;
+
+struct jsl_hdmi_pcm {
+	struct list_head head;
+	struct snd_soc_dai *codec_dai;
+	int device;
+};
+
+struct jsl_card_private {
+	struct snd_soc_jack jsl_headset;
+	struct list_head hdmi_pcm_list;
+	struct snd_soc_jack jsl_hdmi[3];
+};
+
+static int sof_da7219_quirk_cb(const struct dmi_system_id *id)
+{
+	sof_da7219_quirk = (unsigned long)id->driver_data;
+	return 1;
+}
+
+static const struct dmi_system_id sof_da7219_quirk_table[] = {
+	{
+		.callback = sof_da7219_quirk_cb,
+		.matches = {
+			DMI_MATCH(DMI_SYS_VENDOR, "Intel Corporation"),
+			DMI_MATCH(DMI_PRODUCT_NAME, "Ice Lake Client"),
+		},
+		.driver_data = (void *)(0),
+	},
+	{}
+};
```

Comment:
i don't think you really need this if you use the board id and use different aliases for jsl and icl drivers.

Reply:
MCLK is different for icl and jsl, so the quirk is used to program da7219 with different freq, but it may make more sense to support jsl alone for now and add icl support when future needs arise.